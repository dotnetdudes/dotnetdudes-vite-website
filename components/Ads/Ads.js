import AdsTemplate from "./template";

// <Ads></Ads>
class Ads extends HTMLElement {
  constructor() {
    super();
  }
  
  connectedCallback() {
    this.innerHTML = AdsTemplate;
  }

}

window.customElements.define('dudes-ads', Ads);