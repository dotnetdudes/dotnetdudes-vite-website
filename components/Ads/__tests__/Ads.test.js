import { describe, test, expect } from "vitest";
// eslint-disable-next-line no-unused-vars
import Ads from "../Ads.js";

describe("Ads static component", () => {
  // Test case 1: Verify that the component renders correctly
  test("Ads component should render the AdsTemplate", () => {
    document.body.innerHTML = `<dudes-ads></dudes-ads>`;
    expect(document.body.innerHTML).toContain("dotnetdudes");
  });
  
  // Test case 2: Verify that the component renders correctly
  test("Ads component should contain text React", () => {
    document.body.innerHTML = `<dudes-ads></dudes-ads>`;
    expect(document.body.innerHTML).toContain("React");
  });
  
  // Test case 3: Verify that the component renders correctly
  test("Ads component should contain text React Native", () => {
    document.body.innerHTML = `<dudes-ads></dudes-ads>`;
    expect(document.body.innerHTML).toContain("React Native");
  });
  
  // snapshot testing
  test("Ads component should match current snaapshot", () => {
    document.body.innerHTML = `<dudes-ads></dudes-ads>`;
    expect(document.body.innerHTML).toMatchSnapshot();
  });
});