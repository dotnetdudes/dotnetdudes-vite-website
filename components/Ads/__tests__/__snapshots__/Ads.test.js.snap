// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Ads component should render the AdsTemplate 1`] = `
"<dudes-ads>
<div>
    <svg class="dudes-ads dudes-apps">
        <use xlink:href="/icons/install_mobile_FILL0_wght200_GRAD0_opsz48.svg#mobile" href="/icons/install_mobile_FILL0_wght200_GRAD0_opsz48.svg#mobile"></use>
    </svg>
    <h2>Our Apps</h2> 
    <p>
        Check out some of our home grown software solutions. Designed and developed in house  
        by dotnetdudes. Arcu odio ut sem nulla pharetra diam sit amet nisl. Mauris 
        pharetra et ultrices neque ornare. Scelerisque purus semper eget duis at tellus. 
        Condimentum lacinia quis vel eros donec. Viverra suspendisse potenti nullam ac tortor. 
        Lacus suspendisse faucibus interdum posuere. Sed risus ultricies tristique nulla aliquet. 
        Adipiscing enim eu turpis egestas pretium aenean pharetra magna ac. Quis risus sed 
        vulputate odio ut enim blandit volutpat maecenas. Ullamcorper eget nulla 
        facilisi etiam dignissim diam quis enim. Nibh nisl condimentum id venenatis a. 
        Velit sed ullamcorper morbi tincidunt ornare massa eget.
    </p>
</div>
<div>
    <svg class="dudes-ads starter">
        <use xlink:href="/icons/rocket_launch_FILL0_wght200_GRAD0_opsz48.svg#rocket" href="/icons/rocket_launch_FILL0_wght200_GRAD0_opsz48.svg#rocket"></use>
    </svg>
    <h2>React Starter Kits</h2>
    <p>
        Get your React and React Native projects off to a jump start and at the same time
        help to enforce best practices and consistency. Each starter kit can be cloned from our repo to 
        provide an instant, fully functional and full featured application and development environment.
    </p>
    <p>
        Increase overall productivity and quality. Reduce the amount of boilerplate you have 
        to write every time you start a new project.
    </p>
    <p>
        Start customising your application and design immediately.
    </p>
</div>
<div>
    <svg class="dudes-ads blog">
        <use xlink:href="/icons/sticky_note_2_FILL0_wght200_GRAD0_opsz48.svg#stickyNote" href="/icons/sticky_note_2_FILL0_wght200_GRAD0_opsz48.svg#stickyNote"></use>
    </svg>
    <h2>Dotnetdudes blog</h2> 
    <p>Development is fun, that's an important thing to remember. This is our attempt to 
        share our interests, hobbies,experience and knowledge. Web and mobile development predominantly. 
        Focusing on better tools, methodologies, and practices within software development. Lots of resources 
        around HTML, CSS and Web components as well as libraries like React and React Native. 
    </p>
    <p>With new technologies flooding in every day, we'll try to keep up with all the updates. 
    </p>
    <p>
        Providing a  resource for keeping up with recent innovations, communicating with fellow 
        developers, and getting a feel of what software development is all about. 
    </p>
</div>
</dudes-ads>"
`;

exports[`Ads static component > Ads component should match current snaapshot 1`] = `
"<dudes-ads>
<div>
    <svg class="dudes-ads dudes-apps">
        <use xlink:href="/icons/install_mobile_FILL0_wght200_GRAD0_opsz48.svg#mobile" href="/icons/install_mobile_FILL0_wght200_GRAD0_opsz48.svg#mobile"></use>
    </svg>
    <h2>Our Apps</h2> 
    <p>
        Check out some of our home grown software solutions. Designed and developed in house  
        by dotnetdudes. Arcu odio ut sem nulla pharetra diam sit amet nisl. Mauris 
        pharetra et ultrices neque ornare. Scelerisque purus semper eget duis at tellus. 
        Condimentum lacinia quis vel eros donec. Viverra suspendisse potenti nullam ac tortor. 
        Lacus suspendisse faucibus interdum posuere. Sed risus ultricies tristique nulla aliquet. 
        Adipiscing enim eu turpis egestas pretium aenean pharetra magna ac. Quis risus sed 
        vulputate odio ut enim blandit volutpat maecenas. Ullamcorper eget nulla 
        facilisi etiam dignissim diam quis enim. Nibh nisl condimentum id venenatis a. 
        Velit sed ullamcorper morbi tincidunt ornare massa eget.
    </p>
</div>
<div>
    <svg class="dudes-ads starter">
        <use xlink:href="/icons/rocket_launch_FILL0_wght200_GRAD0_opsz48.svg#rocket" href="/icons/rocket_launch_FILL0_wght200_GRAD0_opsz48.svg#rocket"></use>
    </svg>
    <h2>React Starter Kits</h2>
    <p>
        Get your React and React Native projects off to a jump start and at the same time
        help to enforce best practices and consistency. Each starter kit can be cloned from our repo to 
        provide an instant, fully functional and full featured application and development environment.
    </p>
    <p>
        Increase overall productivity and quality. Reduce the amount of boilerplate you have 
        to write every time you start a new project.
    </p>
    <p>
        Start customising your application and design immediately.
    </p>
</div>
<div>
    <svg class="dudes-ads blog">
        <use xlink:href="/icons/sticky_note_2_FILL0_wght200_GRAD0_opsz48.svg#stickyNote" href="/icons/sticky_note_2_FILL0_wght200_GRAD0_opsz48.svg#stickyNote"></use>
    </svg>
    <h2>Dotnetdudes blog</h2> 
    <p>Development is fun, that's an important thing to remember. This is our attempt to 
        share our interests, hobbies,experience and knowledge. Web and mobile development predominantly. 
        Focusing on better tools, methodologies, and practices within software development. Lots of resources 
        around HTML, CSS and Web components as well as libraries like React and React Native. 
    </p>
    <p>With new technologies flooding in every day, we'll try to keep up with all the updates. 
    </p>
    <p>
        Providing a  resource for keeping up with recent innovations, communicating with fellow 
        developers, and getting a feel of what software development is all about. 
    </p>
</div>
</dudes-ads>"
`;
